pageserver/client/src/page_service.rs

```diff
@@ -108,9 +108,32 @@ pub struct RelTagBlockNo {
 }
 
 impl PagestreamClient {
-    pub async fn shutdown(mut self) {
-        let _ = self.cancel_on_client_drop.take();
-        self.conn_task.await.unwrap();
+    pub async fn shutdown(self) {
+        let Self {
+            copy_both,
+            cancel_on_client_drop: cancel_conn_task,
```

Comment:
I think it's easier to follow if this weren't renamed.

Reply:
I'm planning to rename it to `cancel_conn_task` at some point, that name is more precise.